charms/kfp-api/tests/integration/test_charm.py

```diff
@@ -211,6 +153,9 @@ async def test_relational_db_relation_with_mysql_k8s(self, ops_test: OpsTest):
         )
         assert ops_test.model.applications[APP_NAME].units[0].workload_status == "active"
 
+    # FIXME: this test case belongs in unit tests as it is asserting the status of the
+    # unit under a certain condition, we don't actually need the presence of any deployed
+    # charm to test this.
     @pytest.mark.abort_on_fail
```

Comment:
Strongly disagree here. We must have deployed charm to test the integration, because the data supplied by mysql charm can be invalid. In fact, we already had the situation like this and this integration test helped to uncover it.